crypto/x509/x509_cmp.c

```diff
@@ -250,16 +250,20 @@ int X509_NAME_cmp(const X509_NAME *a, const X509_NAME *b)
     return ret < 0 ? -1 : ret > 0;
 }
 
-unsigned long X509_NAME_hash(const X509_NAME *x)
+unsigned long X509_NAME_hash_ex(const X509_NAME *x, int *ok)
```

Comment:
I again apologize for being so demanding, but when you're adding a new _ex function anyway it would be a good idea to add libctx and propq arguments to the function allowing fetching the digest properly.

I am also not sure if returning the error by means of output argument will not be frowned upon by other commiters. I am personally OK with that as otherwise it would complicate changing the callers of the legacy function.

Reply:
> I again apologize for being so demanding, but when you're adding a new _ex function anyway it would be a good idea to add libctx and propq arguments to the function allowing fetching the digest properly.

I like the idea of further improving `X509_NAME_hash_ex()` this way - done.

> I am also not sure if returning the error by means of output argument will not be frowned upon by other commiters. I am personally OK with that as otherwise it would complicate changing the callers of the legacy function.

I see little other choice than adding such an output parameter since with 32-bit `long` there is no room in the return value space that could be used to indicate failure.